ckan/lib/uploader.py

```diff
@@ -218,22 +218,22 @@ def upload(self, id, max_size=10):
                     raise
             tmp_filepath = filepath + '~'
             with open(tmp_filepath, 'wb+') as output_file:
-                with self.upload_file:
-                    self.upload_file.seek(0)
-                    current_size = 0
-                    while True:
-                        current_size = current_size + 1
-                        # MB chunks
-                        data = self.upload_file.read(2 ** 20)
-
-                        if not data:
-                            break
-                        output_file.write(data)
-                        if current_size > max_size:
-                            os.remove(tmp_filepath)
-                            raise logic.ValidationError(
-                                {'upload': ['File upload too large']}
-                            )
+                self.upload_file.seek(0)
+                current_size = 0
+                while True:
+                    current_size = current_size + 1
+                    # MB chunks
+                    data = self.upload_file.read(2 ** 20)
+
+                    if not data:
+                        break
+                    output_file.write(data)
+                    if current_size > max_size:
+                        os.remove(tmp_filepath)
+                        raise logic.ValidationError(
+                            {'upload': ['File upload too large']}
+                        )
+                self.upload_file.close()
                 os.rename(tmp_filepath, filepath)
```

Comment:
This is a sticky comment but why not break it out int the following?
```
def _write_file(self, filepath):
    tmp_filepath = filepath + '~'
    with open(tmp_filepath, 'wb+') as output_file:
        self.upload_file.seek(0)
        current_size = 0
        while True:
            current_size = current_size + 1
            # MB chunks
            data = self.upload_file.read(2 ** 20)
            if not data:
                break
            output_file.write(data)
            if current_size > max_size:
                os.remove(tmp_filepath)
                raise logic.ValidationError(
                    {'upload': ['File upload too large']}
                )
        self.upload_file.close()
        os.rename(tmp_filepath, filepath)
```
then you could just call self._write_file(filename) that's my two cents

Reply:
Unless I'm missing something, I don't think this changes the logic at all, just refactors the existing logic into a method. However, I think there are two problems with this:

1. This is based on a cherry-pick of an upstream commit, so we're trying to keep it as close to upstream as possible.
2. If this function isn't likely to be used anywhere else, making it a separate function doesn't gain anything.

Reply:
I agree with point 1. For point 2: Dan and I both really like the principles described in the book "Clean Code". It talks about having code read like English and about making sub-methods whenever possible with a carefully chosen name. So the purpose for creating the sub-function is not reusability but instead readability and maintainability. Reading this book has really changed my approach to coding and I highly recommend it.

Reply:
In general I agree with using sub-methods to make code more readable, and I often do that myself. However, I'm not sure that it adds anything here, as the code is quite tightly coupled with the calling context. Adding another method increases noise and clutter, and sometimes a one-line comment is a simpler way to achieve the objective.

However, point (1) makes it moot.

Reply:
All of these point I agree with, if we're doing it upstream let's just keep to their logic.

It was a suggestion:)